### PR TITLE
added profile check before publishing apis

### PR DESCRIPTION
**Pull request by ayyoob**

### CODE REVIEW

File: components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/config/Profiles.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @XmlElement(name = "Profile")
+    protected List<String> profile;
+
+    /**
+     * Gets the value of the profile property.
+     *
+     */
+    public List<String> getProfile() {
+        if (profile == null) {
+            profile = new ArrayList<String>();
```
Comment (prabathabey) [inline components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/config/Profiles.java:52]: Make these variables get initialized when the class is loaded so you can then get rid of the null checks. That said, I wouldn't drop the fact that tactics similar to what you've used are generally employed when initializing heavy-weight objects thereby allowing the code to lazy-initialize objects, which is not the case in this particular instance.

**Comment by ayyoob** [comment]
fixed with https://github.com/wso2/carbon-device-mgt/pull/332/commits/892d5cee205400fa936baf05b5e21e020c34f5d9

File: components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/config/Profiles.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+
```
Comment (mohanvive) [inline components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/config/Profiles.java:1]: Please add the license header..

Comment (ayyoob) [reply components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/config/Profiles.java:1]: fixed with https://github.com/wso2/carbon-device-mgt/pull/403/commits/3140c36f7b80a1bfe81be075f84809e9b8cddc3e

**Pull request merged**